src/datasource.ts

```diff
@@ -233,10 +241,12 @@ export class DataSource extends DataSourceApi<SumoQuery> implements DataSourceWi
           const firstKeyedError = keyedErrors.find((keyedError) => keyedError.values?.type === 'error');
           const firstWarningError = keyedErrors.find((keyedError) => keyedError.values?.type === 'warning');
 
-          throw new Error(mapKeyedErrorMessage(firstKeyedError ?? firstWarningError));
+          const errorOrWarning = firstKeyedError ?? firstWarningError;
+          const rowId = errorOrWarning?.values?.rowId;
+          throw new Error(mapKeyedErrorMessage(firstKeyedError ?? firstWarningError, rowId));
         }
 
-        const data = response.flatMap(({ rowId, results }) => {
+        const data = [...response, ...rowsWithNoData].flatMap(({ rowId, results }) => {
```

Comment:
Won't rows with no data be duplicated now? They're already present in `response`, right?

Reply:
They are not presented in response. That's why error messages for them were ignored.